bridge-fungible-token/src/utils.sw

```diff
@@ -92,18 +122,23 @@ pub fn adjust_deposit_decimals(msg_val: b256) -> Result<u64, BridgeFungibleToken
         if decimal_diff > 19u8 {
             return Result::Err(BridgeFungibleTokenError::BridgedValueIncompatability);
         };
-        let adjustment_factor = 10.pow(LAYER_1_DECIMALS - DECIMALS);
-        let bn_factor = U256::from((0, 0, 0, adjustment_factor));
-        let adjusted = value.divide(bn_factor);
 
-        let result = shift_decimals_left(adjusted, decimal_diff);
+        let adjustment_factor = U256::from((0, 0, 0, 10u32.pow(LAYER_1_DECIMALS - DECIMALS)));
+        let result = shift_decimals_right(value, decimal_diff);
+
         if result.is_err() {
             return Result::Err(BridgeFungibleTokenError::BridgedValueIncompatability);
         };
 
-        if result.unwrap() == value
-            && (value > bn_factor
-            || value == bn_factor)
+        let (adjusted, remainder) = result.unwrap();
+        // ensure that the value does not use higher precision than is bridgeable by this contract
+        if remainder != 0u32 {
+            return Result::Err(BridgeFungibleTokenError::BridgedValueIncompatability);
+        }
+
+        // ensure that the value is large enough to be bridged
+        if (value > adjustment_factor
```

Comment:
This check is no longer needed. If `value` is less than the `adjustment_factor` it gaurentees that `shift_decimals_right` will result in a remainder, which will result in an error before this line is even hit.

Reply:
I will check again, but initial tests where I removed this check and set decimals to be >= 19 showed this to be still needed.